pmlc/dialect/pxa/transforms/stencil_xsmm.cc

```diff
@@ -0,0 +1,203 @@
+    int64_t tiles = 1;
+    for (unsigned i = 0; i < semanticIdxCount; i++) {
+      tiles *= llvm::divideCeil(getIdxRange(perm.indexes[i]), tileSize[i]);
+    }
+    return tiles;
+  }
+
+  void transform(TensorAndIndexPermutation perm, ArrayRef<int64_t> tileSize) {
+    // TODO: Clean up this logging
+    IVLOG(2, "Best Perf: " << bestCost);
+    IVLOG(2, "Best Tensor/Index Permutations: TODO: print");
+    std::stringstream bestTilingStr;
+    bestTilingStr << "[ ";
+    for (const auto &tileSize : bestTiling) {
+      bestTilingStr << tileSize << " ";
+    }
+    bestTilingStr << "]";
+    IVLOG(2, "Best Tiling: " << bestTilingStr.str());
+
+    op.setAttr("is_gemm", mlir::UnitAttr::get(op.getContext()));
+  }
+
+public:
+  explicit StencilXSMM(mlir::AffineParallelOp op) : StencilGeneric{op} {
+    // TODO ctor
+    // TODO: Probably want to move these to be params on StencilGeneric ctor...
+    semanticIdxCount = 3; // TODO [i.e., must match generators & requirements]
+    requirements =        // TODO: Make nicer
+        std::map<std::pair<int64_t, int64_t>,
+                 std::function<bool(mlir::Operation *, mlir::BlockArgument)>>{
+            {{0, 0},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] != 0;
+             }},
+            {{0, 1},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] == 0;
+             }},
+            {{0, 2},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] == 1;
+             }},
+            {{1, 0},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] == 0;
+             }},
+            {{1, 1},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] == 1;
+             }},
+            {{1, 2},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] != 0;
+             }},
+            {{2, 0},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] != 0;
+             }},
+            {{2, 1},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] == 1;
+             }},
+            {{2, 2},
+             [this](mlir::Operation *rawOp, mlir::BlockArgument a) {
+               return getStrideInfo(rawOp)->strides[a] == 0;
+             }},
+        };
+    tilingGenerators.push_back(PowerOfTwoGenerator());
+    tilingGenerators.push_back(PowerOfTwoGenerator());
+    tilingGenerators.push_back(PowerOfTwoGenerator());
+  }
+};
+
+struct XSMMStencilPass
+    : public mlir::PassWrapper<XSMMStencilPass, mlir::FunctionPass> {
```

Comment:
BTW, to use the new `passes.td`, replace the base class with the generated one from tablegen: `NewXSMMStencilBase`